[ISSUE] Issue with `databricks_grant` resource on external_location introduced on 1.48.0

### Configuration
We're trying to apply grants to an external location
```hcl
resource "databricks_grants" "main" {
  external_location = "external1"
  id                = (known after apply)

  grant {
      principal  = "GROUP-NAME"
      privileges = [
          "ALL_PRIVILEGES",
        ]
    }
}

```

### Expected Behavior
Grants should be applied

### Actual Behavior
Grants aren't applied with it saying it can't find the group by name. It doesn't matter if the group name is specified as lower or upper case, the error always shows lower case.  We were wondering if it had anything to do with the case sensitivity being removed in 1.48.0.

```
module.external_location_quality.databricks_grants.main: Creating...
2024-06-25T14:17:23.215-0500 [ERROR] provider.terraform-provider-databricks_v1.48.0: Response contains error diagnostic: @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/diag/diagnostics.go:58 tf_provider_addr=registry.terraform.io/databricks/databricks diagnostic_detail="" diagnostic_severity=ERROR diagnostic_summary="cannot create grants: Could not find principal with name group-name" tf_proto_version=5.6 tf_req_id=37a13376-0db4-4936-971b-eeffbb094e31 tf_resource_type=databricks_grants @module=sdk.proto tf_rpc=ApplyResourceChange timestamp=2024-06-25T14:17:23.215-0500
2024-06-25T14:17:23.215-0500 [ERROR] vertex "module.external_location_quality.databricks_grants.main" error: cannot create grants: Could not find principal with name group-name
╷
│ Error: cannot create grants: Could not find principal with name group-name
│
│   with module.external_location_quality.databricks_grants.main,
│   on ../../modules/databricks-external-location/main.tf line 21, in resource "databricks_grants" "main":
│   21: resource "databricks_grants" "main" {
│
```

### Steps to Reproduce
1. `terraform apply` the above code with provider version 1.48.0.  

### Is it a regression?
yes, 1.47.0 and 1.46.0 both work properly.

--- a/catalog/resource_grant.go
+++ b/catalog/resource_grant.go
@@ -20,15 +20,15 @@ func diffPermissionsForPrincipal(principal string, desired catalog.PermissionsLi
 	// diffs change sets for principal
 	configured := map[string]*schema.Set{}
 	for _, v := range desired.PrivilegeAssignments {
-		if strings.EqualFold(v.Principal, principal) {
-			configured[strings.ToLower(v.Principal)] = permissions.SliceToSet(v.Privileges)
+		if v.Principal == principal {
+			configured[v.Principal] = permissions.SliceToSet(v.Privileges)
 		}
 	}
 	// existing permissions that needs removal for principal
 	remote := map[string]*schema.Set{}
 	for _, v := range existing.PrivilegeAssignments {
-		if strings.EqualFold(v.Principal, principal) {
-			remote[strings.ToLower(v.Principal)] = permissions.SliceToSet(v.Privileges)
+		if v.Principal == principal {
+			remote[v.Principal] = permissions.SliceToSet(v.Privileges)
 		}
 	}
 	// STEP 1: detect overlaps
@@ -87,7 +87,7 @@ func filterPermissionsForPrincipal(in catalog.PermissionsList, principal string)
 	grantsForPrincipal := []permissions.UnityCatalogPrivilegeAssignment{}
 	for _, v := range in.PrivilegeAssignments {
 		privileges := []string{}
-		if strings.EqualFold(v.Principal, principal) {
+		if v.Principal == principal {
 			for _, p := range v.Privileges {
 				privileges = append(privileges, p.String())
 			}
@@ -122,7 +122,7 @@ func parseSecurableId(d *schema.ResourceData) (string, string, string, error) {
 func ResourceGrant() common.Resource {
 	s := common.StructToSchema(permissions.UnityCatalogPrivilegeAssignment{},
 		func(m map[string]*schema.Schema) map[string]*schema.Schema {
-			common.CustomizeSchemaPath(m, "principal").SetForceNew().SetCustomSuppressDiff(common.EqualFoldDiffSuppress)
+			common.CustomizeSchemaPath(m, "principal").SetForceNew()
 
 			// set custom hash function for privileges
 			common.MustSchemaPath(m, "privileges").Set = func(i any) int {

--- a/catalog/resource_grant_test.go
+++ b/catalog/resource_grant_test.go
@@ -538,7 +538,7 @@ func TestResourceGrantPermissionsList_Diff_CaseSensitive(t *testing.T) {
 		catalog.PermissionsList{ // config
 			PrivilegeAssignments: []catalog.PrivilegeAssignment{
 				{
-					Principal:  "a",
+					Principal:  "A",
 					Privileges: []catalog.Privilege{"a"},
 				},
 				{
@@ -550,7 +550,7 @@ func TestResourceGrantPermissionsList_Diff_CaseSensitive(t *testing.T) {
 		catalog.PermissionsList{
 			PrivilegeAssignments: []catalog.PrivilegeAssignment{ // platform
 				{
-					Principal:  "A",
+					Principal:  "a",
 					Privileges: []catalog.Privilege{"a"},
 				},
 				{
@@ -560,7 +560,7 @@ func TestResourceGrantPermissionsList_Diff_CaseSensitive(t *testing.T) {
 			},
 		},
 	)
-	assert.Len(t, diff, 0)
+	assert.Len(t, diff, 1)
 }
 
 func TestResourceGrantPermissionsList_Diff_ExternallyAddedPriv(t *testing.T) {

--- a/catalog/resource_grants.go
+++ b/catalog/resource_grants.go
@@ -31,12 +31,12 @@ func diffPermissions(pl catalog.PermissionsList, existing catalog.PermissionsLis
 	// diffs change sets
 	configured := map[string]*schema.Set{}
 	for _, v := range pl.PrivilegeAssignments {
-		configured[strings.ToLower(v.Principal)] = permissions.SliceToSet(v.Privileges)
+		configured[v.Principal] = permissions.SliceToSet(v.Privileges)
 	}
 	// existing permissions that needs removal
 	remote := map[string]*schema.Set{}
 	for _, v := range existing.PrivilegeAssignments {
-		remote[strings.ToLower(v.Principal)] = permissions.SliceToSet(v.Privileges)
+		remote[v.Principal] = permissions.SliceToSet(v.Privileges)
 	}
 	// STEP 1: detect overlaps
 	for principal, confPrivs := range configured {

--- a/catalog/resource_grants_test.go
+++ b/catalog/resource_grants_test.go
@@ -437,7 +437,7 @@ func TestPermissionsList_Diff_CaseSensitivePrincipal(t *testing.T) {
 			},
 		},
 	)
-	assert.Len(t, diff, 0)
+	assert.Len(t, diff, 2)
 }
 
 func TestPermissionsList_Diff_LocalRemoteDiff(t *testing.T) {

--- a/internal/acceptance/grant_test.go
+++ b/internal/acceptance/grant_test.go
@@ -104,7 +104,7 @@ func TestUcAccGrant(t *testing.T) {
 	}, step{
 		Template: strings.ReplaceAll(grantTemplate, "%s", "{env.TEST_DATA_SCI_GROUP}"),
 	}, step{
-		Template: strings.ReplaceAll(strings.ReplaceAll(grantTemplate, "ALL_PRIVILEGES", "ALL PRIVILEGES"), `"%s"`, `upper("{env.TEST_DATA_SCI_GROUP}")`),
+		Template: strings.ReplaceAll(strings.ReplaceAll(grantTemplate, "ALL_PRIVILEGES", "ALL PRIVILEGES"), "%s", "{env.TEST_DATA_ENG_GROUP}"),
 	})
 }
 

--- a/internal/acceptance/grants_test.go
+++ b/internal/acceptance/grants_test.go
@@ -110,7 +110,7 @@ func TestUcAccGrants(t *testing.T) {
 	}, step{
 		Template: strings.ReplaceAll(grantsTemplate, "%s", "{env.TEST_DATA_SCI_GROUP}"),
 	}, step{
-		Template: strings.ReplaceAll(strings.ReplaceAll(grantsTemplate, "ALL_PRIVILEGES", "ALL PRIVILEGES"), `"%s"`, `upper("{env.TEST_DATA_SCI_GROUP}")`),
+		Template: strings.ReplaceAll(strings.ReplaceAll(grantsTemplate, "ALL_PRIVILEGES", "ALL PRIVILEGES"), "%s", "{env.TEST_DATA_ENG_GROUP}"),
 	})
 }
 

